service/share/discovery.go

```diff
@@ -65,18 +76,21 @@ func (d *discovery) handlePeerFound(ctx context.Context, topic string, peer peer
 	d.host.ConnManager().TagPeer(peer.ID, topic, peerWeight)
 }
 
-// findPeers starts peer discovery every 30 seconds until peer cache will not reach peersLimit.
-// TODO (@vgonkivs): simplify when https://github.com/libp2p/go-libp2p/pull/1379 will be merged.
+// findPeers starts peer discovery every discoveryInterval until peer cache reaches peersLimit.
 func (d *discovery) findPeers(ctx context.Context) {
-	t := time.NewTicker(interval * 3)
+	if d.peersLimit == 0 {
+		log.Warn("peers limit is set to 0. Skipping discovery...")
+		return
+	}
+	t := time.NewTicker(d.discoveryInterval)
 	defer t.Stop()
 	for {
 		select {
 		case <-ctx.Done():
 			return
 		case <-t.C:
 			// return if limit was reached to stop the loop and finish discovery
-			if d.set.Size() >= peersLimit {
+			if uint(d.set.Size()) >= d.peersLimit {
```

Comment:
Is there a case in which set size would ever exceed `d.peersLimit`?

Reply:
I guess no